File: backend/app/api/v1/games.py
from fastapi import APIRouter, WebSocket

router = APIRouter()


@router.get("/my-history")
def my_game_history() -> dict[str, list]:
    return {"games": []}


@router.get("/{game_id}")
def get_game(game_id: int) -> dict[str, int | str]:
    return {"message": "Game endpoint scaffolded", "game_id": game_id}


@router.get("/{game_id}/moves")
def get_game_moves(game_id: int) -> dict[str, int | list]:
    return {"game_id": game_id, "moves": []}


@router.post("/{game_id}/resign")
def resign_game(game_id: int) -> dict[str, int | str]:
    return {"message": "Resign endpoint scaffolded", "game_id": game_id}


@router.websocket("/ws/games/{game_id}")
async def game_websocket(websocket: WebSocket, game_id: int) -> None:
    await websocket.accept()
    await websocket.send_json({"type": "info", "payload": {"message": "Game WebSocket scaffolded", "game_id": game_id}})
    await websocket.close()

File: frontend/src/api/games.js
import { apiRequest } from "./client.js";

export function joinMatchmaking() {
  return apiRequest("/api/v1/matchmaking/join", { method: "POST" });
}

export function getGame(gameId) {
  return apiRequest(`/api/v1/games/${gameId}`);
}

export function getGameMoves(gameId) {
  return apiRequest(`/api/v1/games/${gameId}/moves`);
}

export function getMyHistory() {
  return apiRequest("/api/v1/games/my-history");
}
